### PR TITLE
Set azure devops variable if any package reports are found and skip copy packages base on that

### LINKED ISSUE #32067: Official build break in libraries: "Unhandled: Not found SourceFolder: F:\workspace\_work\1\s\artifacts\packages"
Official builds have been failing since Friday with errors like this in the Libraries projects:

> Unhandled: Not found SourceFolder: F:\workspace\_work\1\s\artifacts\packages 

Ex: https://dev.azure.com/dnceng/internal/_build/results?buildId=512293&view=results

### PR DESCRIPTION
**Pull request by safern**
Fixes: https://github.com/dotnet/runtime/issues/32067

Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=515211&_a=summary

cc: @dotnet/runtime-infrastructure

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

File: src/libraries/packages.builds [hunk 1]
```diff
@@ -70,4 +70,14 @@
   </Target>
 
   <Import Project="$(MSBuildThisFileDirectory)dir.traversal.targets" />
+
+  <Target Name="SetAzureDevOpsVariableForBuiltPackages"
+          Condition="'$(OfficialBuildId)' != ''"
```
Comment (safern) [inline src/libraries/packages.builds]: Actually this also needs to run on CI as installer needs produced packages to be in the bin output. Will update.

**Comment by jkotas** [comment]
@safern Were the CI failures that this was merged with known issue? They seem to be introduced by your change.

**Comment by safern** [comment]
Hmm, they didn’t look like my change caused them. Sorry about that, was trying to unblock official builds. I’ll take a look and see what happened and fix them.

**Comment by safern** [comment]
@jkotas this doesn't seem to be caused by my change, there are some rolling builds that I found to be hitting the same issue before my change:
https://dev.azure.com/dnceng/public/_build/results?buildId=515340
https://dev.azure.com/dnceng/public/_build/results?buildId=515340&view=ms.vss-test-web.build-test-results-tab&runId=16323584&resultId=153583&paneView=debug
https://github.com/dotnet/runtime/commits/7b587904e43b8ebf025c9079602d19bd189ffd9b

However. I'm queuing up a build without my change to see if indeed that caused it.

**Comment by safern** [comment]
The build that I queue without my change also hit these tests failures:
https://dev.azure.com/dnceng/public/_build/results?buildId=516253&view=ms.vss-test-web.build-test-results-ta

Maybe it was a change in the jit or the runtime that caused this?

**Comment by ViktorHofer** [comment]
@safern why didn't you just check if the packages directory exists?

**Comment by safern** [comment]
Because you still need to do it as a separate step and emit a variable, plus this was already hooked up as part of the build and it was cross platform. If you run a script to check you need a Unix and a Windows step. 

Yeah we could also check if the directory exists via MSBuild, but since PackageReports are the way we test and rely on our builds to see which packages were produced, this was what made the most sense to me.

**Pull request merged**